insights/parsers/proc_stat.py

```diff
@@ -0,0 +1,131 @@
+        ctxt 17852681
+        btime 1542179825
+        processes 19212
+        procs_running 1
+        procs_blocked 0
+        softirq 11867930 1 3501158 6 4705528 368244 0 79 2021509 0 1271405
+
+
+
+    Examples:
+        >>> type(proc_stat)
+        <class 'insights.parsers.proc_stat.ProcStat'>
+        >>> proc_stat.cpu_percentage
+        '6.73%'
+        >>> proc_stat.ctxt
+        17852681
+        >>> proc_stat.btime
+        '1542179825'
+        >>> proc_stat.softirq_total
+        11867930
+        >>> proc_stat.intr_total
+        21359029
+    """
+
+    def parse_content(self, content):
+        self.data = {}
+        for line in get_active_lines(content):
+            key, value = line.split(None, 1)
+            self.data.update({key: value})
+
+    @property
+    def cpu_percentage(self):
+        """
+        The output of a cpu line could contain the following information according to different kernel version. Earlier
+        versions do not have guest, guestnice for this two are involving for virtual machine in some later version.
+        The cpu usage percentage is calculated by idle divide total.
+        - user:   the number of jiffies (1/100 of a second for x86 systems) that the system has been in user mode
+        - nice:   the number of jiffies that the system has been in user mode with low priority (nice)
+        - system: the number of jiffies that the system has been in system mode
+        - idle:   the number of jiffies that the system has been in idle task
+        - iowait: the number of jiffies that the system has been in I/O wait
+        - irq:    the number of jiffies that the system has been in servicing interrupts
+        - softirq:the number of jiffies that the system has been in servicing softirqs
+        - steal:  the number of jiffies that the system involuntary wait
+        - guest:  the number of jiffies that the system is running a normal guest
+        - guestnice: the number of jiffies that the system is running a niced guest
+        """
+
+        if 'cpu' in self.data:
+            cpu_list = list(map(lambda x: int(x), self.data['cpu'].split()))
+            if len(cpu_list) < 7:
+                raise ParseException("Error: Invalid cpu length", self.data['cpu'])
+            cpu_total = float(sum(cpu_list))
+            cpu_idle = float(cpu_list[3])
+            cpu_pct = (cpu_total - cpu_idle) * 100 / cpu_total
+            return "{0:.2f}".format(cpu_pct) + '%'
+        return None
```

Comment:
@shzhou12 since the data necessary to compute these results are known at the time the data is parsed you could calculate these once in `parse_content` instead of each time this property is invoked.  Having the `ParseException` raised here doesn't really make sense anyway, it should be raised in `parse_content`.